Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/PrefetchingLib.java

```diff
@@ -88,43 +98,33 @@
         return extrasMap;
     }
 
-    private PrefetchingLib(int prefetchStrategyNum) {
-        this.prefetchStrategyNum = prefetchStrategyNum;
-        strategyIntent = PrefetchingStrategy.getStrategy(prefetchStrategyNum);
+    private PrefetchingLib() {
     }
 
+    private static PrefetchingLib getInstance() {
+        if (instance == null)
+            instance = new PrefetchingLib();
+        return instance;
+    }
 
-    public static void init(Context context, int prefetchStrategyNum) {
+    @SuppressWarnings("unused")
+    public static void init(Context context, PrefetchingStrategyType prefetchingStrategyType) {
+        init(context, prefetchingStrategyType, new HashMap<>());
+    }
+
+    @SuppressWarnings("unused")
+    public static void init(Context context, PrefetchingStrategyType prefetchingStrategyType, Map<PrefetchingStrategyConfigKeys, Object> config) {
         if (instance == null) {
-            //configuration file for
-            File path = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS);
-            File file = new File(path, "config.txt");
-            String psn = Integer.toString(prefetchStrategyNum);
-            int size;
-            FileInputStream stream = null;
-            try {
-                stream = new FileInputStream(file);
-                try {
-                    psn = "";
-                    while ((size = stream.read()) != -1) {
-                        psn += Character.toString((char) size);
-                    }
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-                try {
-                    stream.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
-            }
```

Comment:
Why so many try/catch blocks T_T
Anyhow, this reads a configuration file from the device downloads directory.
This configuration file contains only the strategy ID.

End users are not interested in configuring the library, apps developers are. Instead of passing the configuration via a file, a better solution is to accept a configuration map as parameter to the init method.